### PR TITLE
RavenDB-5964 Why replication doesn't have IDocumentTombstoneAware?

### PR DESCRIPTION
**Pull request by karmeli87**

### CODE REVIEW

File: src/Raven.Server/Documents/Replication/DocumentReplicationLoader.cs [hunk 1]
```diff
@@ -728,9 +730,44 @@ public void Dispose()
             foreach (var outgoing in _outgoing)
                 ea.Execute(outgoing.Dispose);
 
+            _database.DocumentTombstoneCleaner?.Unsubscribe(this);
+
             ea.ThrowIfNeeded();
         }
 
+       
+        public Dictionary<string, long> GetLastProcessedDocumentTombstonesPerCollection()
```
Comment (ayende) [inline src/Raven.Server/Documents/Replication/DocumentReplicationLoader.cs:739]: Might be just easier to change the tombstone deleter to just accept something that says "across all collections, my latest is X"

File: src/Raven.Server/Documents/Replication/DocumentReplicationLoader.cs [hunk 2]
```diff
@@ -728,9 +730,44 @@ public void Dispose()
...
+        {
+            var result = new Dictionary<string, long>(StringComparer.OrdinalIgnoreCase);
+            if (_outgoing.Count == 0)
+            {
+                return result;
+            }
+            // last know etag from source
+            var minEtag = _outgoing.Select(o => o._lastSentDocumentEtag).Min();
```
Comment (ayende) [inline src/Raven.Server/Documents/Replication/DocumentReplicationLoader.cs:747]: This is wrong. What happens if the remote server has just restarted and isn't online. You are looking only at the _connected_ servers.
Instead, you need to keep track of all of the confirmations that we previously got across all of them.

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/Replication/DocumentReplicationLoader.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Server/Documents/Replication/DocumentReplicationLoader.cs:739]: Maybe support @all_docs? https://github.com/ravendb/ravendb/blob/v4.0/src/Raven.Client/Data/Constants.cs#L113

**Comment by HibernatingRhinosServiceAccount** [comment]
I don't think we actually use this

*Hibernating Rhinos Ltd  *

Oren Eini* l CEO l *Mobile: + 972-52-548-6969

Office: +972-4-622-7811 *l *Fax: +972-153-4-622-7811

**Comment by ppekrol** [comment]
this is why I was suggesting to add it to cover the case that replication requires. If the returned result will be <@all_docs, some_etag> then the cleaner will know how to handle that.

**Comment by ayende** [comment]
Oh, as the special key, yeah, that would work

**Pull request closed without merging**